### PR TITLE
Fix/security rails 3 2 22

### PR DESCRIPTION
**Pull request by ulferts**
Fixes all security issues reported today in:

http://weblog.rubyonrails.org/2015/6/16/Rails-3-2-22-4-1-11-and-4-2-2-have-been-released-and-more/

Please note:
- Although the report states that rails 3.2 is affected by CVE-2015-3226, it apparently is not as the spec passed without any adjustments in the productive code
- I had to update rails by hand as the custom sprockets prevents the usage of bundler
- rails 3.2.22 contains more than just the security fix (when compared to 3.2.21):

> Differently from the Rails 4 releases, Rails 3.2.22 includes all the commits from the 3-2-stable branch. This mean [Sic] that now Rails 3.2 supports Ruby 2.2.
- I pinned rack to 1.4.6. This rack version was not mentioned but I am very glad it exists.

Fixes:
- https://community.openproject.org/work_packages/20469

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://ulferts-fix-security-rails-3-2-22.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://ulferts-fix-security-rails-3-2-22.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: Gemfile [hunk 1]
```diff
@@ -76,6 +76,7 @@ gem 'strong_parameters'
 gem 'delayed_job_active_record', '0.3.3'
 gem 'daemons'
 
+gem 'rack', '~>1.4.6'
```
Comment (myabc) [inline Gemfile]: @ulferts 037c722 does the same thing on the Rails 4 branch, but specifies `rack` in the _security fixes_ commented section.

Comment (ulferts) [inline Gemfile]: @myabc thats probably better, I will move it. Just a hint: 037c722 states to use 1.4.5 but you probably knew this already.

**Comment by ulferts** [comment]
https://community.openproject.org/work_packages/20469

File: Gemfile (hunk 1 above)
Comment (myabc) [inline Gemfile]: @ulferts according to the Security posting, `1.4.5` is the first version that contains the fix. I didn't know there was a `1.4.6` release out (it's not neither mentioned in the changelog nor tagged in the [rack repo](https://github.com/rack/rack)).

File: spec/security/active_support_to_json_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# CVE-2015-3226: https://groups.google.com/forum/#!msg/rubyonrails-security/7VlB_pck3hU/3QZrGIaQW6cJ
+# The report states that 3.2 is affected by the vulnerability. However,
+# the test copied from the rails patch (adapted to rspec) passed without fixes
+# in the productive code.
+#
+# It should be safe to remove this when OP is on rails >= 4.1
+
+require 'spec_helper'
+
+describe ActiveSupport, type: :model do
```
Comment (myabc) [inline spec/security/active_support_to_json_spec.rb]: @ulferts do you need `type: model` here?

**Pull request merged**